src/controllers/oauthClient.js

```diff
@@ -0,0 +1,29 @@
+const Client = require('../models/auth/Client')
+const grants = require('../constants/grants')
+
+const randtoken = require('rand-token')
+
+const oauthClientController = module.exports = {}
+
+oauthClientController.getAll = (req, res, next) => {
+  // sends back array of users, filtered by queries
+  Client.find(req.query)
+    .then(clients => res.status(200).send(clients))
+    .catch(next)
+}
+
+oauthClientController.create = (req, res, next) => {
+  // receives json for user in body
+  // sends back created user
+  const secret = randtoken.generate(16)
+  const newClient = new Client({
+    name: req.body.name,
+    user: req.user.id,
+    secret,
+    grants: [grants.authCode],
```

Comment:
What does this do? (I should defo know that but would appreciate advice if you have worked it out)

Reply:
So the oauth2 has a variety of different flows and grant types and ways to get access tokens - see [here](https://alexbilbie.com/guide-to-oauth-2-grants/)

The auth code flow is the common one that everyone uses (user authorizes, client gets a code, client exchanges code + secret for token), but there are also others like the refresh token grant (user exchanges a refresh token + client_secret for an access token) or implicit grant (which I don't fully understand). Anyway, when we create a client we have to specify the grant types we plan to use with them, in this case, the auth code grant type. I've actually managed to get it to work with a refresh token which we should implement in the future, but we don't need to yet.